include/seahorn/PredicateAbstraction.hh

```diff
@@ -14,6 +14,45 @@
 namespace seahorn
 {
 	using namespace llvm;
+
+	class HornDbModel
```

Comment:
`HornDbModel` should provide construct api such as `addDef (fapp, expr)` that takes an `fapp` and an expression and remembers that `expr` is the definition of the predicate in `fapp`.  Then model converter will be easier to write, no need to break encapsulation.